src/pip/_internal/index/package_finder.py

```diff
@@ -742,7 +753,9 @@ def get_install_candidate(self, link_evaluator, link):
         """
         is_candidate, result = link_evaluator.evaluate_link(link)
         if not is_candidate:
-            if result:
+            if 'requires-python' in result:
+                self._log_skipped_link_rp(result)
+            else:
```

Comment:
This changes behaviour when `result` is falsy.

Reply:
not sure why I deleted the initial `if result:`, will fix.